### ISSUE TITLE
[translation] account for nonstandard LRO status strings

### ISSUE THREAD

**Issue by kristapratico** [Azure/azure-sdk-for-python#17344]
The service will return "Cancelled" (with two "l"s), "Cancelling", and "ValidationFailed".

Here are the status strings that azure-core polling currently checks to see if the operation has reached a terminal state: https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/core/azure-core/azure/core/polling/base_polling.py#L52-L54

We need to check for these nonstandard statuses or the LRO will run forever.

Options:
1) Simply map the nonstandard statuses to those of the ones checked in the `get_status` [method](https://github.com/Azure/azure-sdk-for-python/blob/246a3382691619c96aed8f0515653c0fcd8df822/sdk/documenttranslation/azure-ai-documenttranslation/azure/ai/documenttranslation/_polling.py#L53):
```
            if status:
                if status == "ValidationFailed":
                    return "Failed"
                elif status in ["Cancelled", "Cancelling"]:
                    return "Canceled"
                return status
            else:
                raise BadResponse("No status found in body")
```

2) Define a custom polling method that will inherit from LROBasePolling and override the methods checking for a terminal state like here in Text Analytics: https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_lro.py#L32

I favor option 1, but we might need to re-evaluate if it makes sense with the current error states of the service. E.g. if we map "ValidationFailed" to a status == "Failed", now we will throw an exception instead of just returning the `JobStatusDetail` when `wait_until_done` is called by the user.

**Comment by mohamedshabanofficial** [comment]
i think we should go with the first approach
much simpler

**Comment by kristapratico** [comment]
If we take option 1, then we always throw in the case of a ValidationError. Error cases 4 and 6 on this [table](https://microsoft.sharepoint.com/teams/AzureDeveloperExperience/_layouts/15/Doc.aspx?sourcedoc={5a1e5006-e46b-4a7c-af6b-fbcd0eb1048a}&action=edit&wd=target%28Translation.one%7C065cac03-b1bc-4ef9-ba83-29fdb6308691%2FService%20errors%20table%7C6ee677b3-ca3b-49a4-aa7e-85c404c3f908%2F%29&wdorigin=703) probably shouldn't raise an exception since there isn't any error details at the batch level (user would need to check the documents to understand what happened). I'm going to bring this up in our v-team chat.

**Comment by kristapratico** [comment]
Tentative plan (may/may not make it for beta.1)


Service terminal   status | Job level error? | Document level   error? | SDK behavior
-- | -- | -- | --
ValidationFailed | Yes | No, returns 0   documents in list | Throw exception   with job level error
Failed | No | Yes, per document | Return   JobStatusDetail, user can inquire on each document for error
Succeeded / (maybe   partiallySucceeded) | No | No | Return   JobStatusDetail
Cancelled/Cancelling | No | No | Return nothing

**Comment by mohamedshabanofficial** [comment]
@kristapratico 
can you elaborate on the ValidationFailed, what is this "job level error"?
i mean what type of error should i throw in here:
https://github.com/mohamedshabanofficial/azure-sdk-for-python/blob/fe31acbe58edbbf7eebf87dc4e1896b45fc70af8/sdk/documenttranslation/azure-ai-documenttranslation/azure/ai/documenttranslation/_polling.py#L85

**Comment by kristapratico** [comment]
"Job level error" will be the error returned on the [BatchStatusDetail](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/documenttranslation/azure-ai-documenttranslation/azure/ai/documenttranslation/_generated/models/_models.py#L97). We'll use the error from the generated type (instead of JobStatusDetail) since we'll raise the exception before the code reaches the models wrapping methods. 

In Python, we throw [HttpResponseError](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/core/azure-core/azure/core/exceptions.py#L241) from azure.core.exceptions. We can extract out the error code and message from the returned error and then instantiate and raise the exception like this:

```
error = HttpResponseError(message="({}): {}".format(error["code"], error["message"]), response=pipeline_response)
raise error
```

**Comment by kristapratico** [comment]
closed with #17738
